exporter/importables.go

```diff
@@ -817,4 +817,65 @@ var resourcesMap map[string]importable = map[string]importable{
 			return nil
 		},
 	},
+	"databricks_notebook": {
+		Service: "notebooks",
+		Name: func(d *schema.ResourceData) string {
+			name := d.Get("path").(string)
+			if name == "" {
+				return d.Id()
+			} else {
+				name = strings.TrimPrefix(name, "/")
+			}
+			re := regexp.MustCompile(`[^0-9A-Za-z_]`)
+			return re.ReplaceAllString(name, "_")
+		},
+		List: func(ic *importContext) error {
+			notebooksAPI := workspace.NewNotebooksAPI(ic.Context, ic.Client)
+			notebookList, err := notebooksAPI.List("/", true)
+			if err != nil {
+				return err
+			}
+			// TODO: emit permissions for notebook folders if non-default,
+			// as per-notebook permission entry would be a noise in the state
+			for offset, notebook := range notebookList {
+				ic.Emit(&resource{
+					Resource: "databricks_notebook",
+					ID:       notebook.Path,
+				})
+				if offset%50 == 0 {
+					log.Printf("[INFO] Scanned %d of %d notebooks",
+						offset+1, len(notebookList))
+				}
+			}
+			return nil
+		},
+		Body: func(ic *importContext, body *hclwrite.Body, r *resource) error {
+			notebooksAPI := workspace.NewNotebooksAPI(ic.Context, ic.Client)
+			status, err := notebooksAPI.Read(r.ID)
+			if err != nil {
+				return err
+			}
+			contentB64, err := notebooksAPI.Export(r.ID, "SOURCE")
```

Comment:
I would prefer to try to export as DBC first as it will keep results & things like dashboards/graphs that are lost when you use SOURCE mode.

Reply:
but how much DBC exports is helping with migration to TF?.. the other concern is granularity of DBC - per user? what about the non-/Users folders, that are sometimes used?

Reply:
You can get individual notebooks as DBC

Reply:
Maybe export format should be configurable?

Reply:
if we're solving the problem of "re-architect this workspace for disaster recovery" or "migrate ST or PVC into MT" problems via TF, then we should provide good use-case by use-case migration tactics. If notebooks are considered application code, then it should be exported as source. repos are for everything else.

DBC is only meant for exporting bulks of notebooks, like Databricks Academy or Spark Summit cases.